payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -418,83 +433,29 @@ internal class PaymentSheetViewModel internal constructor(
     internal class Factory(
         private val applicationSupplier: () -> Application,
         private val starterArgsSupplier: () -> PaymentSheetContract.Args,
+        private val activityLauncherFactory: ActivityLauncherFactory,
+        private val authHostSupplier: () -> AuthActivityStarter.Host,
+        private val eventReporterSupplier: (() -> EventReporter)? = null
     ) : ViewModelProvider.Factory {
         override fun <T : ViewModel?> create(modelClass: Class<T>): T {
-            val application = applicationSupplier()
-            val config = PaymentConfiguration.getInstance(application)
-            val publishableKey = config.publishableKey
-            val stripeAccountId = config.stripeAccountId
-            val stripeRepository = StripeApiRepository(
-                application,
-                { publishableKey }
-            )
-
-            val starterArgs = starterArgsSupplier()
-
-            val googlePayRepository =
-                starterArgs.config?.googlePay?.environment?.let { environment ->
-                    DefaultGooglePayRepository(
-                        application,
-                        environment
-                    )
-                } ?: GooglePayRepository.Disabled
-
-            val prefsRepository = starterArgs.config?.customer?.let { (id) ->
-                DefaultPrefsRepository(
-                    application,
-                    customerId = id,
-                    isGooglePayReady = { googlePayRepository.isReady().first() },
-                    workContext = Dispatchers.IO
+            return DaggerPaymentSheetViewModelComponent.builder()
```

Comment:
Dagger can inject into Activity  when we have access to `Application`, which will host all instances even the activity is destroyed.(So that the same singleton can be injected to the `Activity` again when it's recreated). 

Here since we don't have access to Application, to make sure all the dependencies are preserved, we can only leverage to the `ViewModel` instance, which is preserved by Android system when `Activity` rotates. 

Please refer to [this](https://paper.dropbox.com/doc/Daggerize-Android-SDK--BMogRp6oi~cizo3ZDN~oMI0MAg-Y8OUkag2eU9iPwUhUaVKU#:uid=065112180127781500147735&h2=The-solution) section on more details